src/test/java/seedu/address/logic/commands/HighImportanceCommandTest.java

```diff
@@ -0,0 +1,99 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.logic.commands.CommandTestUtil.showPersonAtIndex;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_PERSON;
+import static seedu.address.testutil.TypicalPersons.getTypicalAddressBook;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.Person;
+import seedu.address.testutil.PersonBuilder;
+
+class HighImportanceCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_validIndexListView_success() {
+        Person personToEditImportanceStatus = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+        Person highImportancePerson =
+                new PersonBuilder(personToEditImportanceStatus).withHighImportance("true").build();
+        HighImportanceCommand highImportanceCommand = new HighImportanceCommand(INDEX_FIRST_PERSON);
+
+        String expectedMessage =
+                String.format(HighImportanceCommand.MESSAGE_CHANGE_HIGH_IMPORTANCE_SUCCESS, highImportancePerson);
+
+        ModelManager expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.setPerson(personToEditImportanceStatus, highImportancePerson);
+
+        assertCommandSuccess(highImportanceCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidIndexListView_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredPersonList().size() + 1);
+        HighImportanceCommand highImportanceCommand = new HighImportanceCommand(outOfBoundIndex);
+
+        assertCommandFailure(highImportanceCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_validIndexFilteredList_success() {
+        showPersonAtIndex(model, INDEX_FIRST_PERSON);
+
+        Person personToEditImportanceStatus = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+        Person highImportancePerson =
+                new PersonBuilder(personToEditImportanceStatus).withHighImportance("true").build();
+        HighImportanceCommand highImportanceCommand = new HighImportanceCommand(INDEX_FIRST_PERSON);
+
+        String expectedMessage =
+                String.format(HighImportanceCommand.MESSAGE_CHANGE_HIGH_IMPORTANCE_SUCCESS, highImportancePerson);
+
+        Model expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
```

Comment:
I was thinking since expectedModel was used in a few of these test methods, maybe they could be declared as a field of this test class. In the event that you were to add more tests, this might come in handy. Just a little suggestion, but there's no need to make any changes! Great job on the tests though!

Reply:
Thanks for the suggestion, will definitely keep that in mind next time to reduce repetition!